### ISSUE TITLE
Should use python3 for src/bully.py

### ISSUE THREAD

**Issue by rhertzog** [aanarchyy/bully#21]
Python 2.x will no longer be supported by their upstream developers in 2020. Thus Debian developers are actively removing Python 2 support in Debian Testing with the goal of getting rid of Python 2 in Debian 11 (bullseye). 

It would be nice if src/bully.py could be ported to Python 3 so that we can get rid of python 2 usage in bully. Please update the shebang line to indicate that Python 3 is preferred.

Actually looking at the code, it seems to have provisions to run with Python 3... unfortunately, it needs to import "bully_support" but that module is only available as pre-compiled Python 2 bytecode (src/bully_support.pyc). Please add the sourcce code for that module.

Thank you very much.

**Comment by kimocoder** [comment]
2to3 src/bully.py

Python 2 is moving into deprecation of year 2020.

**Comment by rhertzog** [comment]
@kimocoder But we can't do that on src/bully_support.py because that file is just not here...

**Comment by kimocoder** [comment]
It's better of the official dev does it, unfortunately most bully forks is unmaintained these days.. aanarchy's, wiire-a's etc. But let's hope for the best :+1:

**Comment by soxrok2212** [comment]
There’s not much support for WPS any more. Especially with WPA3’s debut, it will be replaced with the new Device Provisioning Protocol (DPP) which should take the stage instead.

**Comment by kimocoder** [comment]
Affirmative. Thanks @soxrok2212 -- notify @re4son @aircrack-ng

**Comment by ghost** [comment]
@soxrok2212 but WPS attack is not dead, have a lot of vulnerable routers, brute, pixie, etc, the time to replace all vulnerable routers can be very long, I think bully need an update, maybe an small update, just to add "null pin attack" this python update etc, thinking like you, then put in the trash wifite, aircrack, because wpa3 is the future... An far future! At least 20 years! An bully update to next 20 years is not valid ?

**Comment by soxrok2212** [comment]
WPA3 is already in active deployment, at least in the US. Almost every ISP here has WPS disabled by default now.

**Comment by ghost** [comment]
@soxrok2212, yes here too, but, some countries have a lot of old vulnerable routers running, may need a lot of years to replace all this routers, believe in me, US is an exception, an bully update can be used for many years yet, may be not in US, but at least in "some countries" can! I think is much valid an small update, to add "null pin" and others small things...

**Comment by kimocoder** [comment]
@soxrok2212 is correct. Just added support for WPA3-SAE to rtl8812au.

Unfortunately for them, the WPA3 flaws (side channel attacks) was patched, but the patch went wrong and instead opened another attack. (reported by Mathy Vanhoef)

**Comment by ghost** [comment]
the wifi developers are genius... @kimocoder ,  please read my last post, with arguments

**Comment by kimocoder** [comment]
Ooh yes I agree, here in Norway lots of routers still are vulnarable to pixie (and WPS) :+1: 
I've seen increase in lock out the past 3 years, but still.. working good here

**Comment by ghost** [comment]
@kimocoder you have intention to update your bully fork? Adding null pin etc?

**Comment by kimocoder** [comment]
That's the one thing missing yes, but I'm busy with NetHunter again these days

**Comment by kimocoder** [comment]
@mrochaaa @rhertzog 

Not trying to be rude here, hijacking this thread, but both my bully fork and wifite2 fork is way ahead of the future.. derv82 (wifite project) is gone again, and bully.. I used the wiire-a fork over anarchyy's. It needed a some major improvements, so added lots of fixes to it, also pixiewps and a wordgen support.

And the wifite2 fork has lots of new features, it may run on total automation (except 2 bugs that needs to be addressed now) also, I also made wifite2 run of python3 today.

I'll email derv82 about permission granting to participate on the repo, so we may actually use it in the future also. Enjoy

**Comment by ghost** [comment]
@kimocoder, very good!
